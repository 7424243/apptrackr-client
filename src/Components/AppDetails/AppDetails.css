a {
    text-decoration: none;
    color: #e27802;
}

/* Classes */

.appdetails {
    margin-left: 20px;
    margin-right: 20px;
    max-width: 100%;
    margin-bottom: 25px;
    text-align: left;
    font-size: large;
}

.appdetail_button {
    text-decoration: none;
    margin-right: 20px;
}

.appdetails_buttons {
    display: flex;
    flex-direction: column;
    margin-left: 20px;
}

.appdetails_container {
    display: flex;
    flex-direction: column;
    padding-bottom: 40px;
}

.appdetails_header {
    margin-left: 20px;
    text-align: left;
    padding-top: 20px;
}

.appdetails_small_container {
    margin-right: 20px;
    max-width: 100%;
    text-align: left;
    padding-bottom: 40px;
    font-size: larger;
}

/* Media Query */

@media only screen and (min-width: 425px) {
    .appdetails_buttons {
        flex-direction: row;
    }
}

@media only screen and (min-width: 700px) {

    .appdetails {
        text-align: left;
    }

    .appdetails_buttons {
        justify-content: center;
    }

    .appdetails_header {
        text-align: center;
    }

    .appdetails_small_container {
        width: 60%;
        margin: auto;
    }

}

