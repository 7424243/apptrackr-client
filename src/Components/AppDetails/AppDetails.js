import React, {Component} from 'react'
import RecButton from '../RecButton/RecButton'
import './AppDetails.css'
import ApptrackrContext from '../../ApptrackrContext'
import { NavLink } from 'react-router-dom'
import config from '../../config'
import TokenService from '../../services/token-service'
import { format } from 'date-fns';
import { confirmAlert } from 'react-confirm-alert'
import 'react-confirm-alert/src/react-confirm-alert.css'
import PropTypes from 'prop-types'

class AppDetails extends Component {

    state = {
        error: null
    }

    static contextType = ApptrackrContext

    handleClickDelete = e => {
        e.preventDefault()
        const id = parseInt(this.props.match.params.id)
        console.log(id)
        confirmAlert({
            title: 'Confirm to Delete',
            message: 'Are you sure you want to delete this?',
            buttons: [
                {
                    label: 'Yes',
                    onClick: () => {
                        fetch(`${config.API_ENDPOINT}/applications/${id}`, {
                            method: 'DELETE',
                            headers: {
                                'Authorization': `bearer ${TokenService.getAuthToken()}`
                            }
                        })
                            .then(res => {
                                if(!res.ok) {
                                    return res.json().then(e => Promise.reject(e))
                                }
                                return res
                            })
                            .then(() => {
                                this.context.deleteApplication(id)
                                this.props.history.push('/jobapps')
                            })
                            .catch(error => {
                                this.setState({error})
                                console.error({error})
                            })
                    }
                },
                {
                    label: 'No',
                    onClick: () => this.props.history.push(`/jobapps/${id}`)
                }
            ]
        })
    }

    render() {
        const applicationId = parseInt(this.props.match.params.id)
        const {applications} = this.context
        const getApplication = (applications, applicationId) => 
            applications.find(application => application.id === applicationId)
        const applicationDetails = getApplication(applications, applicationId)
        return (
            <>
                <div className='appdetails_container'>
                    <h2 className='appdetails_header'>{applicationDetails ? applicationDetails.job_name : null}</h2>
                    <div className='appdetails_small_container'>
                     
                            <p className='appdetails'><strong>Company:</strong> {applicationDetails ? applicationDetails.company_name : null}</p>
                            <p className='appdetails'><strong>Website:</strong> {applicationDetails && applicationDetails.website_url ? <a classname='website' href={applicationDetails.website_url} target='_blank' rel='noreferrer'>Go To Website â‡¢</a> : null}</p>
                            <p className='appdetails'><strong>Date Applied:</strong> {applicationDetails && applicationDetails.date_applied ? format(new Date(applicationDetails.date_applied), 'MM/dd/yyyy') : null}</p>
                            <p className='appdetails'><strong>Contact:</strong> {applicationDetails ? applicationDetails.contact_name : null}</p>
                            <p className='appdetails'><strong>Contact Email:</strong> {applicationDetails ? applicationDetails.contact_email : null}</p>
                            <p className='appdetails'><strong>Contact Phone:</strong> {applicationDetails ? applicationDetails.contact_phone : null}</p>
                            <p className='appdetails'><strong>Interview Date:</strong> {applicationDetails && applicationDetails.interview_date ? format(new Date(applicationDetails.interview_date), 'MM/dd/yyyy') : null}</p>
  
                            <p className='appdetails'><strong>Status:</strong> {applicationDetails ? applicationDetails.status : null}</p>
                            <p className='appdetails'><strong>Additional Notes:</strong> {applicationDetails ? applicationDetails.notes : null}</p>

                    </div>
                    
                </div>
                <div className='appdetails_buttons'>
                    <NavLink to={'/jobapps'} className='appdetail_button'><RecButton>Back</RecButton></NavLink>
                    <NavLink to={`/edit/${applicationId}`} className='appdetail_button'><RecButton>Edit</RecButton></NavLink>
                    <NavLink to={'/jobapps'} className='appdetail_button'><RecButton type='button' onClick={this.handleClickDelete}>Delete</RecButton></NavLink>
                </div>
            </>
        )
    }
}

export default AppDetails

AppDetails.propTypes = {
    history: PropTypes.shape({
        push: PropTypes.func
    }).isRequired,
    match: PropTypes.shape({
        params: PropTypes.shape({
            id: PropTypes.string
        })
    }).isRequired
}