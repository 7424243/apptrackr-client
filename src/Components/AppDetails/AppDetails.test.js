import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import AppDetails from './AppDetails'

describe('AppDetails Component', () => {

    const props = {
        match: {
            params: {
                id: '2'
            }
        },
        history: {
            push: jest.fn()
        }
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><AppDetails {...props}/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><AppDetails {...props}/></BrowserRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})