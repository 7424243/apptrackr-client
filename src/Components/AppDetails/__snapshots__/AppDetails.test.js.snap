// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDetails Component renders the UI as expected 1`] = `
Array [
  <div
    className="appdetails_container"
  >
    <h2
      className="appdetails_header"
    />
    <div
      className="appdetails_small_container"
    >
      <p
        className="appdetails"
      >
        <strong>
          Company:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Website:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Date Applied:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Contact:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Contact Email:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Contact Phone:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Interview Date:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Status:
        </strong>
         
      </p>
      <p
        className="appdetails"
      >
        <strong>
          Additional Notes:
        </strong>
         
      </p>
    </div>
  </div>,
  <div
    className="appdetails_buttons"
  >
    <a
      aria-current={null}
      className="appdetail_button"
      href="/jobapps"
      onClick={[Function]}
    >
      <div
        className="recbutton_content"
      >
        <button
          className="recbutton"
        >
          Back
        </button>
      </div>
    </a>
    <a
      aria-current={null}
      className="appdetail_button"
      href="/edit/2"
      onClick={[Function]}
    >
      <div
        className="recbutton_content"
      >
        <button
          className="recbutton"
        >
          Edit
        </button>
      </div>
    </a>
    <a
      aria-current={null}
      className="appdetail_button"
      href="/jobapps"
      onClick={[Function]}
    >
      <div
        className="recbutton_content"
      >
        <button
          className="recbutton"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </a>
  </div>,
]
`;
